pkg/provider/vault/vault.go

```diff
@@ -182,15 +183,17 @@ func (v *client) readSecret(ctx context.Context, path, version string) (map[stri
 
 	secretData := vaultSecret.Data
 	if v.store.Version == esv1alpha1.VaultKVStoreV2 {
+
 		// Vault KV2 has data embedded within sub-field
 		// reference - https://www.vaultproject.io/api/secret/kv/kv-v2#read-secret-version
 		dataInt, ok := vaultSecret.Data["data"]
+
 		if !ok {
-			return nil, errors.New(errVaultData)
+			return nil, errors.New(fmt.Sprintf("failed to find data field: %v", vaultSecret.Data))
 		}
 		secretData, ok = dataInt.(map[string]interface{})
 		if !ok {
-			return nil, errors.New(errVaultData)
+			return nil, errors.New(fmt.Sprintf("failed to unmarshall JSON: %v", dataInt))
```

Comment:
Please create a new declared error string on top for this one, just to follow the pattern